### PR TITLE
RFC for `bundle canonical`

### PR DESCRIPTION
**Pull request by agrim123**
[Rendered](https://github.com/bundler/rfcs/blob/94f5eb38121bda0f56600d3f36f024a39421239b/text/0000-bundle-canonical.md)

### CODE REVIEW

File: text/0000-bundle-canonical.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+```ruby
+# Gemfile after
+
+source "https://rubygems.org/"
+
+# a modular Ruby webserver interface
+gem "rack"
+```
+
+Adding `--view` option does not change the Gemfile rather shows what the Gemfile would look like if run without `--view`.
```
[the remark this answers is not in the document]
Comment (agrim123) [reply text/0000-bundle-canonical.md]: `--dry-run` would be fine, I think

File: text/0000-bundle-canonical.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+$ bundle canonical
+```
+
+```ruby
+# Gemfile after
+
+source "https://rubygems.org/"
+
+# a modular Ruby webserver interface
+gem "rack"
```
[the remark this answers is not in the document]
Comment (agrim123) [reply text/0000-bundle-canonical.md]: Currently, it only works well for a standard Gemfile.

**Comment by deivid-rodriguez** [comment]
I'm not sure this is worth adding to `bundler`. As I stated in bundler's PR

> I'm unsure about this change, rubocop already provides a bunch of style rules for Gemfiles, and people caring about style & prettyness are probably using rubocop already. For those people, this new command will probably not be useful since it's very likely that it won't be able to achieve the level of flexibility rubocop already has. So in my opinion any efforts towards standarizing Gemfile's appearance should be focused on improving existing and adding new cops for rubocop.

To elaborate a bit further on this: Given the level of configurabily `rubocop` has, it seems unlikely that what bundler considers "canonical" exactly matches what rubocop considers "canonical". That means `bundle canonical` and `rubocop --auto-correct Gemfile` might give different results and compete/conflict with each other. And that might result in rubocop users not using `bundle canonical` because of this. And those are the potential users interested in this feature...

**Comment by deivid-rodriguez** [comment]
We should probably come up with a decision here.

**Comment by indirect** [comment]
I'm happy to leave this to `ruby cop --auto-correct Gemfile`. 👍

**Pull request closed without merging**